src/components/Input/index.tsx

```diff
@@ -19,7 +19,13 @@ export type Filter = {
   test: RegExp | Function;
 };
 
-export interface Props {
+export type ControlledInputProps<T> = {
+  value?: T;
+  onChange: (value?: T, event?: any) => void;
+  onStateChange?: (state: boolean) => void;
```

Comment:
@higorquintao I see you're using this callback to communicate whether the input is valid or not, but when reading "onStateChange" I would expect to receive the actual input state as the argument, and its value. 

Can we change the name of the function? 
A suggestion would be `onValidationChange?: (isValid: boolean)`